_includes/footer.html

```diff
@@ -70,9 +79,54 @@ <h2>Follow Us</h2>
 <!-- Google tag (gtag.js) -->
 <script async src="https://www.googletagmanager.com/gtag/js?id=G-3V6VPT445S"></script>
 <script>
+  const gtagId = "{{ site.gtagId }}";
   window.dataLayer = window.dataLayer || [];
   function gtag(){dataLayer.push(arguments);}
-  gtag('js', new Date());
 
-  gtag('config', 'G-3V6VPT445S');
+  // Consent Mode setup
+  gtag("consent", "default", {
+    ad_storage: "denied",
+    analytics_storage: "denied",
+  });
+  $(document).ready(function () {
+    // Check if the consent has been previously given
+    if (!localStorage.getItem("consentGiven")) {
+      $("#consent-banner").show();
```

Comment:
Use variables and re-use in the code to avoid re-create the object on every action. 
For example: 
```
const CONSENT = "consentGiven";
const $consentBanner = jQuery('#consent-banner');
.... 
if (!localStorage.getItem(CONSENT)) {
     $consentBanner.show();
}
```